pkg/controller/networkpolicyrecommendation/controller.go

```diff
@@ -206,6 +215,95 @@ func (c *NPRecommendationController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// handleStaleResources handles the stale Spark Applications and database entries.
+// It will delete the dangling resources without a matching NetworkPolicyRecommendation
+// and add the running NetworkPolicyRecommendation back to the periodical watch list.
+func (c *NPRecommendationController) handleStaleResources() error {
```

Comment:
I kind of think that we should separate sync / retry of the two things we're doing there. Currently populating resync won't be done if errors occur when deleting stale SA or db entries.

Reply:
Currently `handleStaleResources` will be retried as I use a `gcworker` to run it and add back the key upon error. But it's true if deleting stale SA or db entries never succeeds then populating resync will not be executed. Would you prefer to add another worker to take care of the populating resync thing?

Reply:
Or we can persist which part of GC has successfully finished in `gcworker`, and only retry the part that has not. Also, if everything is done, should `gcworker` quit rather than blocking on the queue (returning false on L299)?

Reply:
Sounds great, updated!